src/components/calcite-shell-panel/calcite-shell-panel.tsx

```diff
@@ -86,25 +142,222 @@ export class CalciteShellPanel {
+      collapsed,
+      detached,
+      position,
+      initialContentWidth,
+      contentWidth,
+      contentWidthMax,
+      contentWidthMin,
+      intlResize,
+      resizable
+    } = this;
 
     const contentNode = (
-      <div class={{ [CSS.content]: true, [CSS.contentDetached]: detached }} hidden={collapsed}>
+      <div
+        class={{ [CSS.content]: true, [CSS.contentDetached]: detached }}
+        hidden={collapsed}
+        ref={this.storeContentEl}
+        style={resizable && contentWidth ? { width: `${contentWidth}px` } : null}
+      >
         {this.renderHeader()}
         <div class={CSS.contentBody}>
           <slot />
         </div>
       </div>
     );
 
+    const separatorNode = resizable ? (
+      <div
+        aria-label={intlResize}
+        aria-orientation="horizontal"
+        aria-valuemax={contentWidthMax}
+        aria-valuemin={contentWidthMin}
+        aria-valuenow={contentWidth ?? initialContentWidth}
+        class={CSS.separator}
+        onKeyDown={this.separatorKeyDown}
+        ref={this.connectSeparator}
+        role="separator"
+        tabIndex={0}
+        touch-action="none"
+      />
+    ) : null;
+
     const actionBarNode = <slot name={SLOTS.actionBar} />;
 
-    const mainNodes = [actionBarNode, contentNode];
+    const mainNodes = [actionBarNode, contentNode, separatorNode];
 
     if (position === "end") {
       mainNodes.reverse();
     }
 
     return <div class={{ [CSS.container]: true }}>{mainNodes}</div>;
   }
+
+  // --------------------------------------------------------------------------
+  //
+  //  private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  setContentWidth(width: number): void {
+    const { contentWidthMax, contentWidthMin, position } = this;
+
+    const rounded = Math.round(position === "end" ? width : width);
+
+    this.contentWidth =
+      typeof contentWidthMax === "number" && typeof contentWidthMin === "number"
+        ? clamp(rounded, contentWidthMin, contentWidthMax)
+        : rounded;
+  }
+
+  updateAriaValues = (): void => {
+    const { contentEl } = this;
+    const computedStyle = contentEl && getComputedStyle(contentEl);
+
+    if (!computedStyle) {
+      return;
+    }
+
+    const max = parseInt(computedStyle.getPropertyValue("max-width"), 10);
+    const min = parseInt(computedStyle.getPropertyValue("min-width"), 10);
+    const valueNow = parseInt(computedStyle.getPropertyValue("width"), 10);
+
+    if (typeof valueNow === "number" && !isNaN(valueNow)) {
+      this.initialContentWidth = valueNow;
+    }
+
+    if (typeof max === "number" && !isNaN(max)) {
+      this.contentWidthMax = max;
+    }
+
+    if (typeof min === "number" && !isNaN(min)) {
+      this.contentWidthMin = min;
+    }
+
+    forceUpdate(this);
+  };
+
+  storeContentEl = (contentEl: HTMLDivElement): void => {
+    this.contentEl = contentEl;
+  };
+
+  getKeyvalue = (event: KeyboardEvent): number => {
```

Comment:
Nitpick: `getKeyValue`

Reply:
Nitpick: The event is only passed for the key, so maybe it could be passed directly?

Reply:
Its passed to preventDefault() as well.

Reply:
Ah, never mind then. 👍🏼